R/utils.R

```diff
@@ -139,3 +139,34 @@ check_package_available <- function(pkg) {
     stop("Please install suggested package: ", pkg)
   }
 }
+
+# hidden, also in hemibrainr
+add_field_seq <- function(x, entries, field = "bodyid", ...){
+  x = nat::as.neuronlist(x)
+  if(length(entries)!=length(x)){
+    stop("The length of the entries to add must be the same as the length of the neuronlist, x")
+  }
+  nl = nat::neuronlist()
+  for(i in 1:length(x)){
+    y = x[[i]]
+    entry = entries[i]
+    y[[field]] = entry
+    nl = c(nl, nat::as.neuronlist(y))
+  }
+  names(nl) = names(x)
+  nl[,] = x[,]
+  nl
+}
+
+# hidden
+nullToZero <- function(x) {
+  if(is.list(x)){
+    x[sapply(x, is.null)] <- 0
```

Comment:
why do you want to get a list back here? In my experience it is common to have a list in a dataframe column that you want to have as a vector.

Reply:
if x is already a list. This function is similar to a hemibrainr::nullToNA, where I occasionally needed a list back